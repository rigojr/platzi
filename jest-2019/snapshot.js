export const getCharacter = (data) => ({
    "id": data.id,
    "name": data.name,
    "status": data.species,
    "gender":  data.gender
})