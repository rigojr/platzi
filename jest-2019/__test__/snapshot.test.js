import { getCharacter } from '../snapshot';
import rick from '../rick.json';

describe('Snapshots testing', () => {

    it('first test', () => {
        expect(getCharacter(rick))
            .toMatchSnapshot();
    });

    it('testing execption', () => {
        const user = {
            createAt: new Date(),
            id: Math.floor(Math.random() * 10000),
        };
        expect(user).toMatchSnapshot({
            createAt: expect.any(Date),
            id: expect.any(Number)
        });
    });

    it(' should have exception', () => {
        const user = {
            id: Math.floor(Math.random() * 10000),
            name: "Jose"
        };
        expect(user).toMatchSnapshot({
            id: expect.any(Number)
        });
    });

});
