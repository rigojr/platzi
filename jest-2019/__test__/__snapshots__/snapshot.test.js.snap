// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots testing  should have exception 1`] = `
Object {
  "id": Any<Number>,
  "name": "Jose",
}
`;

exports[`Snapshots testing first test 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanches",
  "status": "Human",
}
`;

exports[`Snapshots testing testing execption 1`] = `
Object {
  "createAt": Any<Date>,
  "id": Any<Number>,
}
`;
